### ISSUE TITLE
4 fingers workspace change don't work with this extension

### ISSUE THREAD

**Issue by BenjaminSchubert** [zakkak/workspace-grid-gnome-shell-extension#21]
When using gnome-wayland, you can have 4 fingers mouse gestures to switch workspaces (swipe UP and swipe down). However, when having this extension enabled, I get : 

```
Gjs-WARNING **: JS ERROR: Exception in callback for signal: activated: Error: Expected type enum for Argument 'direction' but got type 'object'
calculateWorkspace@/home/tellendil/.local/share/gnome-shell/extensions/workspace-grid@mathematical.coffee.gmail.com/extension.js:259
overrideKeybindingsAndPopup/WMProto.actionMoveWorkspace@/home/tellendil/.local/share/gnome-shell/extensions/workspace-grid@mathematical.coffee.gmail.com/extension.js:537
WindowManager<._actionSwitchWorkspace@resource:///org/gnome/shell/ui/windowManager.js:940
wrapper@resource:///org/gnome/gjs/modules/lang.js:169
_emit@resource:///org/gnome/gjs/modules/signals.js:124
TouchpadWorkspaceSwitchAction<._checkActivated@resource:///org/gnome/shell/ui/windowManager.js:506
wrapper@resource:///org/gnome/gjs/modules/lang.js:169
TouchpadWorkspaceSwitchAction<._handleEvent@resource:///org/gnome/shell/ui/windowManager.js:523
wrapper@resource:///org/gnome/gjs/modules/lang.js:169
```

I couldn't figure out the problem, but if anyone has a clue, I can do more testing.

Thanks !

**Comment by zakkak** [comment]
I tried it on Archlinux with Gnome 3.18.2 and xorg-server-wayland 1.18.0.  I do not get the errors you mention but I can confirm that it is not working.

**Comment by BenjaminSchubert** [comment]
Sorry, I indeed forgot to give versions : 
I'm using gnome-shell 3.18.3 and xorg-server-wayland 1.18.0 underr Fedora 23

**Comment by zakkak** [comment]
Should be fixed in the 3.18 branch, can you please give it a try?

You will need to clone the project and run
```
make zip
```

Then use gnome tweak tool to install the zip.

**Comment by BenjaminSchubert** [comment]
it works perfectly ! thanks a lot !

**Comment by zakkak** [comment]
Fixed in ae637404ba4d7979a58bb037a5a42d5b78a29753
